### PR TITLE
Approve user application

### PR DESCRIPTION
**Pull request by atstojanov**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/cherrytwist/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/cherrytwist/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/cherrytwist/Coordination/blob/master/LICENSE 
     and the contributor license agreement: tba

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ import { ValidationPipe } from '@utils/validation/validation.pipe';
 import oidcConfig from '@config/oidc.config';
 import { ApplicationModule } from '@domain/application/application.module';
 import { NVPModule } from '@domain/nvp/nvp.module';
+import { ApplicationFactoryModule } from '@domain/application/application.factory.module';
```
Comment (techsmyth) [inline src/app.module.ts:44]: So two modules in the one folder?

File: src/domain/application/application.service.ts [hunk 2]
```diff
@@ -35,59 +36,66 @@ export class ApplicationService {
...
-      {
-        id: challengeId,
-      },
-      {
-        relations: ['applications'],
-      }
-    );
-    return challenge?.applications || [];
-  }
+    console.timeEnd('Load application');
```
Comment (techsmyth) [inline src/domain/application/application.service.ts]: left over?

File: src/domain/application/application.service.ts [hunk 3]
```diff
@@ -35,59 +36,66 @@ export class ApplicationService {
...
+    } else if (parent === 'opportunity') {
+      await this.opportunityService.addMember(application.user.id, parentId);
+    }
+
+    application.status = ApplicationStatus.approved;
+
+    await this.applicationReposity.save(application);
+
+    return application;
+    return {} as Application;
```
Comment (techsmyth) [inline src/domain/application/application.service.ts]: left over?

**Review by techsmyth — APPROVED** [review]
Nice update.

Made a bunch of smaller updates as part of the review - please check the last commit. If you are ok with them then please merge through.

File: src/app.module.ts (hunk 1 above)
Comment (atstoyanov) [reply src/app.module.ts:44]: Yes, the factory is used by other service. I don't want to mess services. If we use on module circular dependency is inevitable.

**Pull request merged**